client/pages/upload.jsx

```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+
+export default class Upload extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      imageURL: 'images/image-placeholder.png',
+      condition: '',
+      location: '',
+      price: '',
+      title: '',
+      description: '',
+      token: '',
+      image: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.fileInputRef = React.createRef();
+    this.handleImageSubmit = this.handleImageSubmit.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  }
+
+  handleImageSubmit(event) {
+    const { token } = this.state;
+    event.preventDefault();
+
+    const formData = new FormData();
+    const image = this.fileInputRef.current.files[0];
+    this.setState({ imageURL: image.name });
+    formData.append('image', image);
+
+    fetch('/api/images', {
+      method: 'POST',
+      headers: {
+        'x-access-token': token
+      },
+      body: formData
+    })
+      .then(res => res.json())
+      .then(data => {
+        const { url } = data;
+        this.setState({
+          imageURL: url,
+          image: url
+        });
+        this.fileInputRef.current.value = null;
+      })
+      .catch(err => console.error(err));
+  }
+
+  componentDidMount() {
+    const token = window.localStorage.getItem('lfz-final');
+    this.setState({ token });
+  }
```

Comment:
You can do this in the `handleImageSubmit` directly so that you don't add extra properties to `state`.